lesscpy/lesscpy
# Dump of changes

## Description

This branch ended as a dump of many changes so I will split it in separate problems.

To have a foundation I start writing test for the examples provided by Less documentation.

I will use this branch to try to fix the following errors.
- [x] Lazy evaluation of variables inside a block - https://github.com/less/less.js/blob/master/test/less/lazy-eval.less
- [x] Variables in property names

Variables in import names are not yet implemented... but will do this in another branch.
- [ ] Variables in import names - https://github.com/less/less.js/blob/master/test/less/import-interpolation.less
## Changes

I know that we should use the official test provided by the LESS project, but I find it hard to debug/fix code base on those examples. I find it much easier to be able to write unit test and pin point the error... as the official less files contains a lot of declarations for a single test, each test representing a full less file.

For test test_lazy_loading_scope I think that the error is in upstream documentation so I have changed the expected result.

I have created a dedicated TestCase and in memory error reporter to help with testing. I find the in memory reporter much easier to use than catching  stderror during tests.

I have included the upstream less test files.

For all the errors (regressions) I found while running the less test files I have created a dedicated unit test to pin point the problem. I see that instantiating the parser object takes a lot of time. Maybe we can implement a resetable parser which could be reused between tests.
## How to test/review/open questions

I see that upstream CSS files are using spaces but the files from lesscpy are with tabs. Why is that? I was thinking that it would be good to keep them as close to the original so that in the future there can be an automated script to sync them.

Do we need to run minification test for each less file?

Please let me know if you think that I forgot to add un upstream less test file.

The changes are quite big so I can split this branch in 2.

Thanks!
